connector/src/main/java/org/geysermc/connector/network/translators/block/BlockTranslator.java

```diff
@@ -1,20 +1,141 @@
 package org.geysermc.connector.network.translators.block;
 
+import com.fasterxml.jackson.databind.JsonNode;
 import com.github.steveice10.mc.protocol.data.game.world.block.BlockState;
+import com.nukkitx.nbt.CompoundTagBuilder;
+import com.nukkitx.nbt.NbtUtils;
+import com.nukkitx.nbt.stream.NBTInputStream;
+import com.nukkitx.nbt.tag.CompoundTag;
+import com.nukkitx.nbt.tag.ListTag;
+import gnu.trove.map.TObjectIntMap;
+import gnu.trove.map.hash.TObjectIntHashMap;
+import it.unimi.dsi.fastutil.ints.Int2IntMap;
+import it.unimi.dsi.fastutil.ints.Int2IntOpenHashMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import org.geysermc.connector.console.GeyserLogger;
 import org.geysermc.connector.utils.Toolbox;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.io.InputStream;
+import java.util.*;
 
 public class BlockTranslator {
-    private final Map<String, BlockEntry> javaIdentifierMap = new HashMap<>();
+    public static final ListTag<CompoundTag> BLOCKS;
+    public static final BlockState AIR = new BlockState(0);
 
-    public BlockEntry getBlockEntry(BlockState state) {
-        return Toolbox.BLOCK_ENTRIES.get(state.getId());
+    private static final Int2IntMap JAVA_TO_BEDROCK_BLOCK_MAP = new Int2IntOpenHashMap();
+    private static final Int2ObjectMap<BlockState> BEDROCK_TO_JAVA_BLOCK_MAP = new Int2ObjectOpenHashMap<>();
+    private static final Int2IntMap JAVA_TO_BEDROCK_LIQUID_MAP = new Int2IntOpenHashMap();
+    private static final Int2ObjectMap<BlockState> BEDROCK_TO_JAVA_LIQUID_MAP = new Int2ObjectOpenHashMap<>();
```

Comment:
These Int2ObjectMaps with block states can just be an Int2IntMap because BlockState only has one field in it which is the ID of the java block.

Reply:
You end up sending the `BlockState` so mapping directly would reduce the overhead of creating new instances each time since they are immutable.

Reply:
Gotcha.